src/Ethernet.h

```diff
@@ -219,6 +219,11 @@ class EthernetClient : public Client {
 	uint8_t status();
 	virtual int connect(IPAddress ip, uint16_t port);
 	virtual int connect(const char *host, uint16_t port);
+	// 20190421 https://kmpelectronics.eu/ Plamen Kovandjiev - We added support for ESP32.
+#ifdef ESP32
+	virtual int connect(IPAddress ip, uint16_t port, int timeout);
```

Comment:
We already have a mechanism for setting the timeout:
https://www.arduino.cc/en/Reference/EthernetClientSetConnectionTimeout